src/fides/api/migrations/hash_migration_job.py

```diff
@@ -0,0 +1,102 @@
+from loguru import logger
+from sqlalchemy import text
+from sqlalchemy.orm import Session
+
+from fides.api.db.base_class import FidesBase
+from fides.api.migrations.hash_migration_mixin import HashMigrationMixin
+from fides.api.migrations.hash_migration_tracker import HashMigrationTracker
+from fides.api.models.privacy_preference import (
+    CurrentPrivacyPreference,
+    PrivacyPreferenceHistory,
+    ServedNoticeHistory,
+)
+from fides.api.models.privacy_request import CustomPrivacyRequestField, ProvidedIdentity
+from fides.api.tasks import DatabaseTask, celery_app
+from fides.api.tasks.scheduled.scheduler import scheduler
+from fides.config import CONFIG
+
+HASH_MIGRATION = "hash_migration"
+MODELS_TO_MIGRATE = [
+    CurrentPrivacyPreference,
+    ProvidedIdentity,
+    CustomPrivacyRequestField,
+    PrivacyPreferenceHistory,
+    ServedNoticeHistory,
+]
+
+
+def initiate_bcrypt_migration_task() -> None:
+    """Initiates scheduler to migrate all non-credential tables using a bcrypt hash to use a SHA-256 hash"""
+
+    if CONFIG.test_mode:
+        return
+
+    assert (
+        scheduler.running
+    ), "Scheduler is not running! Cannot migrate tables with bcrypt hashes."
+
+    logger.info("Initiating scheduler for hash migration")
+    scheduler.add_job(
+        func=bcrypt_migration_task,
+        id=HASH_MIGRATION,
+    )
+
+
+@celery_app.task(base=DatabaseTask, bind=True)
```

Comment:
heh, i do appreciate looking for consistency, but i'm _pretty_ sure that this is rarely/never intended if the function  isn't actually  going to be run as a celery task.

as i look a bit closer into it, i suppose it's _possible_ that this type of usage could be intentional, since we actually leverage a different db engine (our 'task engine') within the `DatabaseTask.get_new_session`...so maybe there's a use case where we'd want to use that engine, even if the functions aren't running as celery tasks? it feels a bit contrived - but i'd like to check with @pattisdr if that's what was intended when adding the decorator to e.g. [this function](https://github.com/ethyca/fides/pull/4760/files#diff-e9b609a6bd089ccdf2444e0784f0f81e6fbcc0380601a583c854ae7a0e420557R179-R180) in the DSR 3.0 refactor - or if that's also just accidental usage!

i do wanna get clear on this, because as your comment makes clear, if we don't start clearing this up, it's only going to cause more confusion as we add more scheduled tasks!

Reply:
Too long ago to recall but I doubt it was intentional, just an easy way to get the session.  I agree we're creating unneccesary confusion here by doing it like this.

Reply:
thanks @pattisdr  - 

in that case @galvana, i'd suggest we remove this. it should be pretty straightforward for us to get a well-handled db session - see an analogous example [here](https://github.com/ethyca/fidesplus/blob/main/src/fidesplus/api/service/discovery/discovery_monitor_executor.py#L101) in the discovery monitor executor (this function is also run as a scheduled task!). that's using a function defined in `fidesplus` to retrieve the db session but we can/should add a `@contextmanager` decorator to our `deps.get_db()` in fides OSS, i think, so it can be used similarly...

Reply:
Thank you two for the input! I've promoted `get_db` to a context manager, let's just make sure all the tests pass